osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -211,6 +209,19 @@ public void Save(BeatmapInfo info, IBeatmap beatmapContent)
                 workingCache.Remove(working);
         }
 
+        protected override void PreUpdate(BeatmapSetInfo item)
+        {
+            base.PreUpdate(item);
+
+            foreach (var info in item.Beatmaps)
```

Comment:
Is there a reason this couldn't just be done in the `Save()` method? Not saying that we necessarily should be, but just curious if there is another (non-save) case that requires it to be here.

Reply:
I was thinking it could be used for updating of beatmaps, but in thinking about it all of that is going to go through `Save()` anyway. I've removed it for now.